frontend/src/domain/WeclomeScreen/Form/Form.tsx

```diff
@@ -0,0 +1,104 @@
+import React from "react";
+import { makeStyles } from "@material-ui/core/styles";
+import Grid from "@material-ui/core/Grid";
+import Button from "@material-ui/core/Button";
+import Divider from "@material-ui/core/Divider";
+import { Input } from "../../../components";
+import { Form as FinalForm, Field } from "react-final-form";
+import { theme as appTheme } from "../../../constants";
+
+const useStyles = makeStyles((theme) => ({
+  label: {
+    color: "inherit",
+    textTransform: "uppercase",
+    fontSize: "1rem",
+  },
+  submit: {
+    color: "inherit",
+    backgroundColor: appTheme.colors.darkOrchid,
+    boxShadow: "none",
+    height: theme.spacing(6),
+    margin: theme.spacing(4, 1, 0, 1),
+    "&:hover": {
+      backgroundColor: appTheme.colors.darkOrchid,
+    },
+  },
+  register: {
+    color: "inherit",
+    borderColor: "white",
+    height: theme.spacing(6),
+    margin: theme.spacing(0, 1),
+  },
+  divider: {
+    width: "100%",
+    backgroundColor: appTheme.colors.ivory,
+    margin: theme.spacing(2, 0),
+  },
+  newCustomer: {
+    margin: theme.spacing(1, 1),
+  },
+}));
+
+type FooterProps = {
+  label: string;
+  title: string;
+  href: string;
+};
+
+type InputProps = {
+  label: string;
+  inputProps?: React.ComponentProps<typeof Input>;
+};
+
+export type FormProps = {
+  submitTitle: string;
+  footer: FooterProps;
+  formElements: Array<InputProps>;
+};
+
+export function Form({ submitTitle, footer, formElements }: FormProps) {
+  const classes = useStyles();
+
+  return (
+    <FinalForm
+      onSubmit={() => {}}
+      render={({ handleSubmit, pristine, submitting }) => (
+        <form onSubmit={handleSubmit}>
+          <Grid container spacing={2}>
+            {formElements.map(({ label, inputProps }) => (
+              <Grid item xs={12} sm={12} key={label}>
+                <Field name={label}>
+                  {({ input }) => (
+                    <>
+                      <label className={classes.label}>{label}</label>
+                      <Input {...input} {...inputProps} />
+                    </>
+                  )}
+                </Field>
+              </Grid>
+            ))}
+            <Button
+              type="submit"
+              fullWidth
+              variant="contained"
+              disabled={submitting || pristine}
```

Comment:
Can we overwrite styles for `disabled` state? The text is hardly readable.

![image](https://user-images.githubusercontent.com/33123587/113544821-4c99d280-95e9-11eb-8d07-70496897d1c9.png)

Reply:
updated the color